src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -77,10 +91,45 @@ public List<RawEntityTypeSummary> getEntityTypeSummary(Set<UUID> entityTypeIds)
       );
   }
 
+  private List<EntityTypeColumn> fetchColumnNamesForCustomFields(UUID entityTypeId) {
+    log.info("Getting derived table name for entity type ID: {}", entityTypeId);
+    String sourceViewName = getEntityTypeDefinition(entityTypeId)
+      .map(EntityType::getSourceView)
+      .orElseThrow(() -> new EntityTypeNotFoundException(entityTypeId));
+    return jooqContext
+      .select(field(REQUIRED_FIELD_NAME), field(REF_ID))
+      .from(sourceViewName)
+      .where(field(WHERE_CLAUSE).eq(CUSTOM_FIELD_TYPE))
+      .fetch()
+      .stream()
+      .map(row -> {
+        Object value = row.get(0);
+        Object extractedRefId = row.get(1);
+        assert value != null;
+        return handleSingleCheckBox(value.toString(), extractedRefId.toString());
+      })
+      .toList();
+  }
+
+  private EntityTypeColumn handleSingleCheckBox(String value, String refId) {
+    ValueWithLabel trueValue = new ValueWithLabel().label("True").value("true");
+    ValueWithLabel falseValue = new ValueWithLabel().label("False").value("false");
+
+    return new EntityTypeColumn()
+      .name(value)
+      .dataType(new BooleanType())
+      .values(List.of(trueValue, falseValue))
+      .visibleByDefault(false)
+      .valueGetter("src_users_users.jsonb -> 'customFields' ->> '" + refId + "'")
```

Comment:
Big red flag here: custom fields aren't necessarily only in the Users module (in practice, they are, but only because no one else has bothered to use them, but they are a general concept in a shared library that could easily be used by any module), so we shouldn't be referring to `src_users_users` by name in code. That should come from an entity type definition or some other kind of config.

Reply:
This will be handled by another ticket: https://folio-org.atlassian.net/browse/MODFQMMGR-155